packages/internal/src/ses-utils.js

```diff
@@ -163,6 +172,61 @@ export const assertAllDefined = obj => {
   }
 };
 
+/**
+ * Attenuate `specimen` to only properties allowed by `permit`.
+ *
+ * @template T
+ * @template {Permit<T>} P
+ * @param {T} specimen
+ * @param {P} permit
+ * @param {<U, SubP extends Permit<U>>(attenuation: U, permit: SubP) => U} [transform]
+ *   used to replace the results of recursive picks (but not blanket permits)
+ * @returns {Attenuated<T, P>}
+ */
+export const attenuate = (specimen, permit, transform = x => x) => {
```

Comment:
this is useful. why bucket it in "SES utils"? I see other stuff in here that's not specific to the `ses` package so not a big deal. Maybe we just rename this module at some point.

Reply:
regardless, needs unit tests.